solr/core/src/test/org/apache/solr/handler/TestHttpRequestId.java

```diff
@@ -126,7 +116,7 @@ public void onFailure(Throwable throwable) {
       assertEquals(3, reqLog.getQueue().size());
       while (!reqLog.getQueue().isEmpty()) {
         var reqEvent = reqLog.getQueue().poll();
-        assertTrue(reqEvent.getContextData().containsKey(key));
+        assertNotNull(reqEvent);
```

Comment:
don't think this change is correct. we are checking if the MDC actually contains the unique key. the log will always be there, so the `assertNotNull` will always hold

Reply:
the check is that reqEvent could be null from the poll(). I'm going to push a quick addendum quick to remove the repeated `reqLog.getQueue` which then doesn't need the null check.

Reply:
Fixed in 117960e0d521c18ce48a7f2b520a65a0eab4870b